src/components/input/Button.tsx

```diff
@@ -27,6 +27,12 @@ type ComponentProps = {
    */
   title?: string;
 
+  /**
+   * Whether the focus ring should be applied to `:focus` or `:focus-visible`.
+   * Defaults to focus-visible.
+   */
+  focusRing?: 'focus' | 'focus-visible';
```

Comment:
> See https://github.com/hypothesis/frontend-shared/pull/1292/files#r1345926735 for more details on why this has been introduced

I think `focusRing` sounds like a boolean property. Perhaps a better name would be one of `focusRingMode`, `focusRingType` or `focusRingVariation`.